tests/integration/tls/test_tls.py

```diff
@@ -154,3 +158,74 @@ async def test_tls_renewal(ops_test: OpsTest) -> None:
+
+    # Deploy TLS Certificates operator
+    config = {"ca-common-name": "CN_CA", "certificate-validity": "1"}
+    await ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="stable", config=config)
+    await wait_until(ops_test, apps=[TLS_CERTIFICATES_APP_NAME], apps_statuses=["active"])
+
+    # Deploy Opensearch operator
+    await ops_test.model.set_config(MODEL_CONFIG)
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.deploy(
+        my_charm,
+        num_units=1,
+        series=SERIES,
+    )
+
+    await wait_until(
+        ops_test,
+        apps=[APP_NAME],
+        units_statuses=["blocked"],
+        wait_for_exact_units=1,
+    )
+
+    # Now apply a hack to make the certificate secrets expire in short time
+    # set the secret expiry to a fixed timedelta of 180 seconds to give time to start initially.
+    # This happens on the tls_certificates lib and we apply the patch via sed-command.
+    # The patch accesses the method `_get_next_secret_expiry_time` in the tls_certificate libs.
+    # If the test should fail on updates to this lib, check if the method was renamed or removed.
+    search_expression = "expire=self._get_next_secret_expiry_time\\(certificate\\)"
+    replace_expression = "expire=timedelta\\(seconds=180\\)"
+
+    unit_id = get_application_unit_ids(ops_test, APP_NAME)[0]
+    unit_ip = await get_leader_unit_ip(ops_test)
+    lib_file = f"/var/lib/juju/agents/unit-opensearch-{unit_id}/charm/lib/charms/tls_certificates_interface/v3/tls_certificates.py"
+    cmd = f"juju ssh {APP_NAME}/{unit_id} sudo sed -i 's/{search_expression}/{replace_expression}/g' {lib_file}"
+    subprocess.run(cmd, shell=True)
+
+    # Relate OpenSearch to TLS and wait until all is settled
+    await ops_test.model.integrate(APP_NAME, TLS_CERTIFICATES_APP_NAME)
+
+    # wait for the unit to be ready and API's available
+    cluster_health_resp = await cluster_health(ops_test, unit_ip, wait_for_green_first=True)
+    assert cluster_health_resp["status"] == "green"
+
+    # now start with the actual test
+    # first get the currently used certs
+    current_certs = await get_loaded_tls_certificates(ops_test, unit_ip)
+
+    # now wait for the expiration period to pass by (and a bit longer for things to settle)
+    # we can't use `wait_until` here because the unit might not be idle in the meantime
+    logger.info("Waiting for certificates to expire.")
+    time.sleep(SECRET_EXPIRY_WAIT_TIME)
+
+    # now compare the current certificates against the earlier ones and see if they were updated
+    updated_certs = await get_loaded_tls_certificates(ops_test, unit_ip)
+    logger.info(f"Certificates before expiry: {current_certs}")
+    logger.info(f"Certificates after expiry: {updated_certs}")
+
+    assert (
+        updated_certs["transport_certificates_list"][0]["not_before"]
+        > current_certs["transport_certificates_list"][0]["not_before"]
+    )
```

Comment:
Could we add 2 additional checks here: 
- if all the nodes are online 
- if the new certificates are effectively loaded / used by all the units

Reply:
Was added.